tools/generate-docs

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash -e
+
+# PRELUDE: FIND THE REAL PATH OF THIS FILE
+pushd . > '/dev/null';
+SCRIPT_PATH="${BASH_SOURCE[0]:-$0}";
+
+while [ -h "$SCRIPT_PATH" ];
+do
+    cd "$( dirname -- "$SCRIPT_PATH"; )";
+    SCRIPT_PATH="$( readlink -f -- "$SCRIPT_PATH"; )";
+done
+
+cd "$( dirname -- "$SCRIPT_PATH"; )" > '/dev/null';
+SCRIPT_PATH="$( pwd; )";
+popd  > '/dev/null';
```

Comment:
It tackles a lof of edgecases: `../script`, `script/abc`, symlink, symlink of a symlnk and more. I've taken it from this reply since some years:

https://stackoverflow.com/a/12197518

Reply:
ok, `readlink` might not be installed everywhere. But OTOH we don't have a use case for symlinking this script here anyway, right?
I wouldn't add code for a use case that we might never have, so maybe just drop the symlink stuff and do `cd "$( dirname "$( dirname "$SCRIPT_PATH" )" )"`

Reply:
Please note, that in the original script, the `cd` is only used to store the real path to the script in to the variable and it is immediately followed by the `popd`.

As seen on line 19 (`cd ./docs/`), the rest of the script expects the project root path as a current working directory (not the `$SCRIPT_PATH` aka `tools/`).

Reply:
Left a comment on the script, and written it so that it assumes it's being called from the makefile or the project's root.

That does make it shorter and simpler :+1: